bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```diff
@@ -0,0 +1,798 @@
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (this.pollingJob != null) {
+            this.pollingJob.cancel(true);
+        }
+        this.comms = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getIdWithoutGroup();
+        if (command instanceof RefreshType) {
+            if (channelId.equals(HeliosEasyControlsBindingConstants.SYS_DATE)) {
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.DATE));
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.TIME));
+            } else if (channelId.equals(HeliosEasyControlsBindingConstants.BYPASS_FROM)) {
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.BYPASS_FROM_DAY));
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.BYPASS_FROM_MONTH));
+            } else if (channelId.equals(HeliosEasyControlsBindingConstants.BYPASS_TO)) {
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.BYPASS_TO_DAY));
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.BYPASS_TO_MONTH));
+            } else {
+                scheduler.submit(() -> readValue(channelId));
+            }
+        } else { // write command
+            String value = null;
+            if (command instanceof OnOffType) {
+                value = command == OnOffType.ON ? "1" : "0";
+            } else if (command instanceof DateTimeType) {
+                ZonedDateTime d = ((DateTimeType) command).getZonedDateTime();
+                if (channelId.equals(HeliosEasyControlsBindingConstants.SYS_DATE)) {
+                    setSysDateTime(d);
+                } else if (channelId.equals(HeliosEasyControlsBindingConstants.BYPASS_FROM)) {
+                    this.setBypass(true, d.getDayOfMonth(), d.getMonthValue());
+                } else if (channelId.equals(HeliosEasyControlsBindingConstants.BYPASS_TO)) {
+                    this.setBypass(false, d.getDayOfMonth(), d.getMonthValue());
+                } else {
+                    value = formatDate(channelId, ((DateTimeType) command).getZonedDateTime());
+                }
+            } else if ((command instanceof DecimalType) || (command instanceof StringType)) {
+                value = command.toString();
+            } else if (command instanceof QuantityType<?>) {
+                // convert item's unit to the Helios device's unit
+                Map<String, HeliosVariable> variableMap = this.variableMap;
+                if (variableMap != null) {
+                    String unit = variableMap.get(channelId).getUnit();
+                    QuantityType<?> val = (QuantityType<?>) command;
+                    if (unit != null) {
+                        if (unit.equals(HeliosVariable.UNIT_DAY)) {

```

Comment:
Can't this be a java switch statement? Same comment for the other large if-else blocks?
```suggestion
                        switch (unit) {
                        case HeliosVariable.UNIT_DAY:
                         ...
```